tests/components/lgthinq/test_config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Test the lgthinq config flow."""
+
+from unittest.mock import AsyncMock, patch
+
+from homeassistant.components.lgthinq.const import (
+    CLIENT_PREFIX,
+    CONF_CONNECT_CLIENT_ID,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_COUNTRY, CONF_NAME, CONF_SOURCE
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .common import mock_thinq_api_response
+from .const import MOCK_COUNTRY, MOCK_ENTRY_NAME, MOCK_PAT
+
+from tests.common import MockConfigEntry
+
+
+async def test_config_flow(hass: HomeAssistant) -> None:
+    """Test that an thinq entry is normally created with valid values."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER}
+    )
+    await hass.async_block_till_done()
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == SOURCE_USER
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_ACCESS_TOKEN: MOCK_PAT,
+            CONF_NAME: MOCK_ENTRY_NAME,
+        },
+    )
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == SOURCE_USER
+
+    with patch("homeassistant.components.lgthinq.config_flow.ThinQApi") as mock:
```

Comment:
Can we create a central patch? Please check `airgradient` for an example. This eliminates any functions improperly patched

Reply:
Refactored